rclcpp/include/rclcpp/topic_statistics/subscription_topic_statistics.hpp

```diff
@@ -172,7 +172,10 @@ class SubscriptionTopicStatistics
   {
     auto received_message_age = std::make_unique<ReceivedMessageAge>();
     received_message_age->Start();
-    subscriber_statistics_collectors_.emplace_back(std::move(received_message_age));
+    {
+      std::lock_guard<std::mutex> lock(mutex_);
+      subscriber_statistics_collectors_.emplace_back(std::move(received_message_age));
+    }
```

Comment:
I actually think we should do this slightly differently.  That's because there is a race between putting the `age` and putting the `period` in place.

I think that what we should do here is to construct both `ReceivedMessageAge` and `ReceivedMessagePeriod`, call `Start()` on each of them, then take the lock and insert both of them into `subscriber_statistics_collectors_`.  That will ensure that they are both in there together.

Reply:
yeah that is better, i will fix that.

Reply:
https://github.com/ros2/rclcpp/pull/2592/commits/69fbd66635dfac2149c715b2f0060487d32e142a addressed it